TST: check inequality by comparing categorical with NaN ( #28384 )

junjunjunk opened the pull request:
- [x] closes #28384
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

junjunjunk committed:
TST: check inequality by comparing categorical with NaN ( #28384 )

junjunjunk committed:
BUG: add comment of issue num

junjunjunk commented:
Please review.

And I need some help.  
I don't know why CI is failing.From the looks of it, it looks like it's timed out.

arw2019 commented on pandas/tests/arrays/categorical/test_missing.py:
```diff
@@ -148,3 +148,13 @@ def test_use_inf_as_na_outside_context(self, values, expected):
             expected = DataFrame(expected)
             tm.assert_frame_equal(result, expected)
+
+    def test_compare_unequal(self):
```
`test_compare_categorical_with_missing`

junjunjunk replied:
Thank you. Fixed.
https://github.com/pandas-dev/pandas/pull/36520/commits/3e293cf0d223322d81a5f6ac632f15973b64335e

arw2019 reviewed (commented):
I think possibly add a non-categorical example since the OP was about the difference in behavior between categorical vs. non-categorical. See https://github.com/pandas-dev/pandas/issues/28384#issuecomment-530514218

Re: CI failures those are unrelated I think. Merge with master and they'll be rerun

junjunjunk committed:
CLN: change test function name

junjunjunk committed:
TST: add test case categorical vs noncategorical

junjunjunk commented:
> I think possibly add a non-categorical example since the OP was about the difference in behavior between categorical vs. non-categorical. See [#28384 (comment)](https://github.com/pandas-dev/pandas/issues/28384#issuecomment-530514218)
> 
> Re: CI failures those are unrelated I think. Merge with master and they'll be rerun

Thank you for your input. I added test case.
And I rebased my branch.

jreback commented on pandas/tests/arrays/categorical/test_missing.py:
```diff
@@ -148,3 +148,22 @@ def test_use_inf_as_na_outside_context(self, values, expected):
             expected = DataFrame(expected)
             tm.assert_frame_equal(result, expected)
+
+    def test_compare_categorical_with_missing(self):
```
can you fully replicate the tests in the OP, e.g. add for == as well.

ideally parameterize this over these 4 cases (or 2 and 2 in-line if easier).

junjunjunk replied:
Thank you for review. Done.

junjunjunk committed:
TST: parametrize & change test case

jreback commented:
thanks @junjunjunk
